Failed to build `tests/subsys/mgmt/mcumgr/os_mgmt_datetime` on `qemu_cortex_m0`

**Describe the bug**
Failed to build `tests/subsys/mgmt/mcumgr/os_mgmt_datetime` on `qemu_cortex_m0`.

https://github.com/zephyrproject-rtos/zephyr/actions/runs/8256335111/job/22585096229?pr=70125 is the CI job which failed.

**To Reproduce**
Steps to reproduce the behavior:
1. `west build -p -b qemu_cortex_m0 tests/subsys/mgmt/mcumgr/os_mgmt_datetime`

**Expected behavior**
No build error.

**Impact**
Annoyance for now as it's failing CI.

**Logs and console output**
```
-- west build: making build dir /tmp/build pristine
-- west build: generating a build system
Loading Zephyr default modules (Zephyr base).
-- Application: $ZEPHYR_BASE/tests/subsys/mgmt/mcumgr/os_mgmt_datetime
-- CMake version: 3.28.3
-- Found Python3: /usr/bin/python3.10 (found suitable version "3.10.13", minimum required is "3.8") found components: Interpreter 
-- Cache files will be written to: $HOME/.cache/zephyr
-- Zephyr version: 3.6.99 ($ZEPHYR_BASE)
-- Found west (found suitable version "1.2.0", minimum required is "0.14.0")
-- Board: qemu_cortex_m0, identifier: nrf51822
-- Found host-tools: zephyr 0.16.5 (/opt/zephyr-sdk/current)
-- Found toolchain: zephyr 0.16.5 (/opt/zephyr-sdk/current)
-- Found Dtc: /opt/zephyr-sdk/current/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.6.0", minimum required is "1.4.6") 
-- Found BOARD.dts: $ZEPHYR_BASE/boards/qemu/cortex_m0/qemu_cortex_m0.dts
-- Found devicetree overlay: $ZEPHYR_BASE/tests/subsys/mgmt/mcumgr/os_mgmt_datetime/boards/qemu_cortex_m0.overlay
-- Generated zephyr.dts: /tmp/build/zephyr/zephyr.dts
-- Generated devicetree_generated.h: /tmp/build/zephyr/include/generated/devicetree_generated.h
-- Including generated dts.cmake file: /tmp/build/zephyr/dts.cmake

warning: QEMU_ICOUNT_SHIFT (defined at boards/Kconfig:62) was assigned the value '6' but got the
value ''. Check these unsatisfied dependencies: QEMU_ICOUNT (=n). See
http://docs.zephyrproject.org/latest/kconfig.html#CONFIG_QEMU_ICOUNT_SHIFT and/or look up
QEMU_ICOUNT_SHIFT in the menuconfig/guiconfig interface. The Application Development Primer, Setting
Configuration Values, and Kconfig - Tips and Best Practices sections of the manual might be helpful
too.

Parsing $ZEPHYR_BASE/Kconfig
Loaded configuration '$ZEPHYR_BASE/boards/qemu/cortex_m0/qemu_cortex_m0_defconfig'
Merged configuration '$ZEPHYR_BASE/tests/subsys/mgmt/mcumgr/os_mgmt_datetime/prj.conf'
Configuration saved to '/tmp/build/zephyr/.config'
Kconfig header saved to '/tmp/build/zephyr/include/generated/autoconf.h'
-- Found GnuLd: /opt/zephyr-sdk/zephyr-sdk-0.16.5/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd (found version "2.38") 
-- The C compiler identification is GNU 12.2.0
-- The CXX compiler identification is GNU 12.2.0
-- The ASM compiler identification is GNU
-- Found assembler: /opt/zephyr-sdk/current/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc
CMake Warning at $ZEPHYR_BASE/CMakeLists.txt:864 (message):
  No SOURCES given to Zephyr library: lib__posix__shell

  Excluding target from build.


-- Configuring done (2.8s)
-- Generating done (0.0s)
-- Build files have been written to: /tmp/build
-- west build: building application
[1/162] Preparing syscall dependency handling

[2/162] Generating include/generated/version.h
-- Zephyr version: 3.6.99 ($ZEPHYR_BASE), build: v3.6.0-840-g3667526b832d
[89/162] Building C object zephyr/soc/soc/nrf51822/CMakeFiles/soc__nordic.dir/validate_base_addresses.c.obj
FAILED: zephyr/soc/soc/nrf51822/CMakeFiles/soc__nordic.dir/validate_base_addresses.c.obj 
/opt/zephyr-sdk/current/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DKERNEL -DK_HEAP_MEM_POOL_SIZE=0 -DNRF51 -DNRF51422_XXAA -DPICOLIBC_LONG_LONG_PRINTF_SCANF -DZCBOR_ASSERTS -D__LINUX_ERRNO_EXTENSIONS__ -D__PROGRAM_START -D__ZEPHYR__=1 -I$ZEPHYR_BASE/include -I/tmp/build/zephyr/include/generated -I$ZEPHYR_BASE/soc/nordic -I$ZEPHYR_BASE/soc/nordic/nrf51/. -I$ZEPHYR_BASE/soc/nordic/common/. -I$ZEPHYR_BASE/subsys/mgmt/mcumgr/util/include -I$ZEPHYR_BASE/subsys/mgmt/mcumgr/transport/include -I$ZEPHYR_BASE/subsys/testsuite/include -I$ZEPHYR_BASE/subsys/testsuite/ztest/include -I<zephyr>/modules/hal/cmsis/CMSIS/Core/Include -I$ZEPHYR_BASE/modules/cmsis/. -I<zephyr>/modules/hal/nordic/nrfx -I<zephyr>/modules/hal/nordic/nrfx/drivers/include -I<zephyr>/modules/hal/nordic/nrfx/mdk -I$ZEPHYR_BASE/modules/hal_nordic/nrfx/. -I<zephyr>/modules/lib/zcbor/include -isystem $ZEPHYR_BASE/lib/libc/common/include -fno-strict-aliasing -Os -imacros /tmp/build/zephyr/include/generated/autoconf.h -fno-printf-return-value -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-m0 -mthumb -mabi=aapcs -mfp16-format=ieee -mtp=soft --sysroot=/opt/zephyr-sdk/current/arm-zephyr-eabi/arm-zephyr-eabi -imacros $ZEPHYR_BASE/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wdouble-promotion -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -ftls-model=local-exec -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=$ZEPHYR_BASE/tests/subsys/mgmt/mcumgr/os_mgmt_datetime=CMAKE_SOURCE_DIR -fmacro-prefix-map=$ZEPHYR_BASE=ZEPHYR_BASE -fmacro-prefix-map=$HOME/zephyr/rtos=WEST_TOPDIR -ffunction-sections -fdata-sections --specs=picolibc.specs -std=c99 -MD -MT zephyr/soc/soc/nrf51822/CMakeFiles/soc__nordic.dir/validate_base_addresses.c.obj -MF zephyr/soc/soc/nrf51822/CMakeFiles/soc__nordic.dir/validate_base_addresses.c.obj.d -o zephyr/soc/soc/nrf51822/CMakeFiles/soc__nordic.dir/validate_base_addresses.c.obj -c $ZEPHYR_BASE/soc/nordic/validate_base_addresses.c
In file included from $ZEPHYR_BASE/include/zephyr/toolchain.h:50,
                 from $ZEPHYR_BASE/include/zephyr/kernel_includes.h:23,
                 from $ZEPHYR_BASE/include/zephyr/kernel.h:17,
                 from $ZEPHYR_BASE/soc/nordic/validate_base_addresses.c:7:
/tmp/build/zephyr/include/generated/devicetree_generated.h:410:37: error: 'DT_N_S_rtc_REG_IDX_0_VAL_ADDRESS' undeclared here (not in a function); did you mean 'DT_N_S_cpus_S_cpu_0_REG_IDX_0_VAL_ADDRESS'?
  410 | #define DT_N_NODELABEL_rtc          DT_N_S_rtc
      |                                     ^~~~~~~~~~
$ZEPHYR_BASE/include/zephyr/toolchain/gcc.h:87:51: note: in definition of macro 'BUILD_ASSERT'
   87 | #define BUILD_ASSERT(EXPR, MSG...) _Static_assert(EXPR, "" MSG)
      |                                                   ^~~~
$ZEPHYR_BASE/include/zephyr/sys/util_internal.h:69:53: note: in expansion of macro '__DEBRACKET'
   69 | #define __GET_ARG2_DEBRACKET(ignore_this, val, ...) __DEBRACKET val
      |                                                     ^~~~~~~~~~~
$ZEPHYR_BASE/include/zephyr/sys/util_internal.h:64:9: note: in expansion of macro '__GET_ARG2_DEBRACKET'
   64 |         __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code)
      |         ^~~~~~~~~~~~~~~~~~~~
$ZEPHYR_BASE/include/zephyr/sys/util_internal.h:59:9: note: in expansion of macro '__COND_CODE'
   59 |         __COND_CODE(_XXXX##_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~
$ZEPHYR_BASE/include/zephyr/sys/util_macro.h:180:9: note: in expansion of macro 'Z_COND_CODE_1'
  180 |         Z_COND_CODE_1(_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~~~
$ZEPHYR_BASE/include/zephyr/sys/util_macro.h:378:23: note: in expansion of macro 'COND_CODE_1'
  378 | #define UTIL_OR(a, b) COND_CODE_1(UTIL_BOOL(a), (a), (b))
      |                       ^~~~~~~~~~~
$ZEPHYR_BASE/soc/nordic/validate_base_addresses.c:93:17: note: in expansion of macro 'UTIL_OR'
   93 |                 UTIL_OR(UTIL_NOT(DT_NODE_EXISTS(DT_NODELABEL(lbl))),    \
      |                 ^~~~~~~
$ZEPHYR_BASE/include/zephyr/devicetree.h:2191:9: note: in expansion of macro 'DT_CAT4'
 2191 |         DT_CAT4(node_id, _REG_IDX_, idx, _VAL_ADDRESS)
      |         ^~~~~~~
$ZEPHYR_BASE/include/zephyr/devicetree.h:2214:30: note: in expansion of macro 'DT_REG_ADDR_BY_IDX'
 2214 | #define DT_REG_ADDR(node_id) DT_REG_ADDR_BY_IDX(node_id, 0)
      |                              ^~~~~~~~~~~~~~~~~~
$ZEPHYR_BASE/soc/nordic/validate_base_addresses.c:94:26: note: in expansion of macro 'DT_REG_ADDR'
   94 |                         (DT_REG_ADDR(DT_NODELABEL(lbl)) == (uint32_t)(mdk_addr))))
      |                          ^~~~~~~~~~~
$ZEPHYR_BASE/include/zephyr/devicetree.h:4451:24: note: in expansion of macro 'DT_N_NODELABEL_rtc'
 4451 | #define DT_CAT(a1, a2) a1 ## a2
      |                        ^~
$ZEPHYR_BASE/include/zephyr/devicetree.h:198:29: note: in expansion of macro 'DT_CAT'
  198 | #define DT_NODELABEL(label) DT_CAT(DT_N_NODELABEL_, label)
      |                             ^~~~~~
$ZEPHYR_BASE/soc/nordic/validate_base_addresses.c:94:38: note: in expansion of macro 'DT_NODELABEL'
   94 |                         (DT_REG_ADDR(DT_NODELABEL(lbl)) == (uint32_t)(mdk_addr))))
      |                                      ^~~~~~~~~~~~
$ZEPHYR_BASE/soc/nordic/validate_base_addresses.c:224:1: note: in expansion of macro 'CHECK_DT_REG'
  224 | CHECK_DT_REG(rtc, NRF_RTC);
      | ^~~~~~~~~~~~
$ZEPHYR_BASE/soc/nordic/validate_base_addresses.c:224:19: error: 'NRF_RTC' undeclared here (not in a function); did you mean 'NRF_RTC0'?
  224 | CHECK_DT_REG(rtc, NRF_RTC);
      |                   ^~~~~~~
$ZEPHYR_BASE/include/zephyr/toolchain/gcc.h:87:51: note: in definition of macro 'BUILD_ASSERT'
   87 | #define BUILD_ASSERT(EXPR, MSG...) _Static_assert(EXPR, "" MSG)
      |                                                   ^~~~
$ZEPHYR_BASE/include/zephyr/sys/util_internal.h:69:53: note: in expansion of macro '__DEBRACKET'
   69 | #define __GET_ARG2_DEBRACKET(ignore_this, val, ...) __DEBRACKET val
      |                                                     ^~~~~~~~~~~
$ZEPHYR_BASE/include/zephyr/sys/util_internal.h:64:9: note: in expansion of macro '__GET_ARG2_DEBRACKET'
   64 |         __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code)
      |         ^~~~~~~~~~~~~~~~~~~~
$ZEPHYR_BASE/include/zephyr/sys/util_internal.h:59:9: note: in expansion of macro '__COND_CODE'
   59 |         __COND_CODE(_XXXX##_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~
$ZEPHYR_BASE/include/zephyr/sys/util_macro.h:180:9: note: in expansion of macro 'Z_COND_CODE_1'
  180 |         Z_COND_CODE_1(_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~~~
$ZEPHYR_BASE/include/zephyr/sys/util_macro.h:378:23: note: in expansion of macro 'COND_CODE_1'
  378 | #define UTIL_OR(a, b) COND_CODE_1(UTIL_BOOL(a), (a), (b))
      |                       ^~~~~~~~~~~
$ZEPHYR_BASE/soc/nordic/validate_base_addresses.c:93:17: note: in expansion of macro 'UTIL_OR'
   93 |                 UTIL_OR(UTIL_NOT(DT_NODE_EXISTS(DT_NODELABEL(lbl))),    \
      |                 ^~~~~~~
$ZEPHYR_BASE/soc/nordic/validate_base_addresses.c:224:1: note: in expansion of macro 'CHECK_DT_REG'
  224 | CHECK_DT_REG(rtc, NRF_RTC);
      | ^~~~~~~~~~~~
$ZEPHYR_BASE/soc/nordic/validate_base_addresses.c:94:25: error: expression in static assertion is not an integer
   94 |                         (DT_REG_ADDR(DT_NODELABEL(lbl)) == (uint32_t)(mdk_addr))))
      |                         ^
$ZEPHYR_BASE/include/zephyr/toolchain/gcc.h:87:51: note: in definition of macro 'BUILD_ASSERT'
   87 | #define BUILD_ASSERT(EXPR, MSG...) _Static_assert(EXPR, "" MSG)
      |                                                   ^~~~
$ZEPHYR_BASE/include/zephyr/sys/util_internal.h:69:53: note: in expansion of macro '__DEBRACKET'
   69 | #define __GET_ARG2_DEBRACKET(ignore_this, val, ...) __DEBRACKET val
      |                                                     ^~~~~~~~~~~
$ZEPHYR_BASE/include/zephyr/sys/util_internal.h:64:9: note: in expansion of macro '__GET_ARG2_DEBRACKET'
   64 |         __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code)
      |         ^~~~~~~~~~~~~~~~~~~~
$ZEPHYR_BASE/include/zephyr/sys/util_internal.h:59:9: note: in expansion of macro '__COND_CODE'
   59 |         __COND_CODE(_XXXX##_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~
$ZEPHYR_BASE/include/zephyr/sys/util_macro.h:180:9: note: in expansion of macro 'Z_COND_CODE_1'
  180 |         Z_COND_CODE_1(_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~~~
$ZEPHYR_BASE/include/zephyr/sys/util_macro.h:378:23: note: in expansion of macro 'COND_CODE_1'
  378 | #define UTIL_OR(a, b) COND_CODE_1(UTIL_BOOL(a), (a), (b))
      |                       ^~~~~~~~~~~
$ZEPHYR_BASE/soc/nordic/validate_base_addresses.c:93:17: note: in expansion of macro 'UTIL_OR'
   93 |                 UTIL_OR(UTIL_NOT(DT_NODE_EXISTS(DT_NODELABEL(lbl))),    \
      |                 ^~~~~~~
$ZEPHYR_BASE/soc/nordic/validate_base_addresses.c:224:1: note: in expansion of macro 'CHECK_DT_REG'
  224 | CHECK_DT_REG(rtc, NRF_RTC);
      | ^~~~~~~~~~~~
[114/162] Building C object modules/hal_nordic/nrfx/CMakeFiles/modules__hal_nordic__nrfx.dir<zephyr>/modules/hal/nordic/nrfx/drivers/src/nrfx_gpiote.c.obj
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /usr/bin/cmake --build /tmp/build
```

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: Zephyr SDK 0.16.5
 - Commit 3667526b832d4925edac085ac30c653b844bd121

**Additional context**
Add any other context that could be relevant to your issue, such as pin setting,
target configuration, ...

--- a/soc/nordic/validate_base_addresses.c
+++ b/soc/nordic/validate_base_addresses.c
@@ -90,7 +90,7 @@
  */
 #define CHECK_DT_REG(lbl, mdk_addr)					\
 	BUILD_ASSERT(							\
-		UTIL_OR(UTIL_NOT(DT_NODE_EXISTS(DT_NODELABEL(lbl))),	\
+		UTIL_OR(UTIL_NOT(DT_REG_HAS_IDX(DT_NODELABEL(lbl), 0)),	\
 			(DT_REG_ADDR(DT_NODELABEL(lbl)) == (uint32_t)(mdk_addr))))
 
 /**

